pkg/agent/controller/servicecidr/provider.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicecidr
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	"antrea.io/antrea/pkg/agent/util"
+)
+
+const (
+	// Disable resyncing.
+	resyncPeriod time.Duration = 0
+)
+
+type EventHandler func()
+
+type Interface interface {
+	GetServiceCIDR(isIPv6 bool) (*net.IPNet, error)
+	// The added handlers will be called when Service CIDR changes.
+	AddEventHandler(handler EventHandler)
+}
+
+type provider struct {
+	serviceInformer cache.SharedIndexInformer
+	sync.RWMutex
+	serviceIPv4CIDR *net.IPNet
+	serviceIPv6CIDR *net.IPNet
+	eventHandlers   []EventHandler
+}
+
+func NewServiceCIDRProvider(serviceInformer coreinformers.ServiceInformer) Interface {
+	p := &provider{
+		serviceInformer: serviceInformer.Informer(),
+	}
+	p.serviceInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    p.addService,
+			UpdateFunc: p.updateService,
+		},
+		resyncPeriod,
+	)
+	return p
+}
+
+func (p *provider) GetServiceCIDR(isIPv6 bool) (*net.IPNet, error) {
+	p.RLock()
+	defer p.RUnlock()
+	if isIPv6 {
+		if p.serviceIPv6CIDR == nil {
+			return nil, fmt.Errorf("Service IPv6 CIDR is not available now")
```

Comment:
probably just say `Service IPv6 CIDR is not available`

Reply:
I prefer adding `now` since it is only not available temperately when starting.

Reply:
Ok, you may say `Service IPv6 CIDR is not available yet for now`, let you to decide.

Reply:
`Service IPv6 CIDR is not available yet for now` is much better

Reply:
I havn't seen a sentence using "yet for now", I think "yet" already means "now" in negative sentences. "Service IPv6 CIDR is not available yet" should be enough